src/main/java/com/dsinnovators/devprofilesbackend/modules/developers/DeveloperService.java

```diff
@@ -115,10 +115,13 @@ public Developer createProfile(Developer developer) {
         return developerRepository.save(developer);
     }
 
-    public Developer updateToken(Long id, String code) throws DeveloperNotFoundException {
+    public Developer updateToken(Long id, String code) throws DeveloperNotFoundException, GithubUserNotFound {
         Developer developer = developerRepository.findById(id)
                                                  .orElseThrow(() -> new DeveloperNotFoundException(id));
         developer.setGithubAccessToken(githubGraphqlClient.getAccessTokenFromGithub(code));
+        Profile existingProfile = developer.githubProfile();
+        developer.getProfiles().remove(existingProfile);
+        developer.addProfile(fetchGithubUserProfile(existingProfile, developer.getGithubAccessToken()));
```

Comment:
Fetching github profile right after updating token will get rid of nullPointerException that occurs after github authentication.